activesupport/test/message_encryptor_test.rb

```diff
@@ -104,6 +104,48 @@ def test_messing_with_aead_values_causes_failures
     assert_aead_not_decrypted(encryptor, [text, iv, auth_tag[0..-2]] * "--")
   end
 
+  def test_expired_messages_raise_error
+    encryptor = ActiveSupport::MessageEncryptor.new(@secret, cipher: "aes-256-gcm")
+    message = { value: "Ruby On Rails", expires_at: 1.year.ago.utc.iso8601 }
+    encrypted_message = encryptor.encrypt_and_sign(message)
+    assert_raise(ActiveSupport::MessageEncryptor::ExpiredMessage) do
+      encryptor.decrypt_and_verify(encrypted_message)
+    end
+  end
+
+  def test_decrypt_and_verify_unexpired_messages
+    encryptor = ActiveSupport::MessageEncryptor.new(@secret, cipher: "aes-256-gcm")
+    message = { value: "Ruby On Rails", expires_at: 1.year.from_now.utc.iso8601 }
+    encrypted_message = encryptor.encrypt_and_sign(message)
+    assert_equal message, encryptor.decrypt_and_verify(encrypted_message)
+  end
+
+  def test_purpose_feature_of_messages
+    encryptor = ActiveSupport::MessageEncryptor.new(@secret, cipher: "aes-256-gcm")
+    message = { value: "Foo", purpose: "bar" }
+    encrypted_message = encryptor.encrypt_and_sign(message)
```

Comment:
This should be `encryptor.encrypt_and_sign(message, purpose: "bar")` and the decrypted message should then just be `"foo"` (remember to keep case consistency too).